### DISCUSSION TITLE
请问为什么第三方库没有用submodule？

### DISCUSSION THREAD

**Discussion by rocketman123456** [BoomingTech/Piccolo#321] [Q&A] [answered]
如题。当需要添加代码量大的大型库时，感觉没有使用submodule不太方便，也没办法方便的更新第三方库到最新版

**Comment by hyv1001** [comment, marked as answer]
最早是用submodule的，但在内部测试的时候发现很多同学在更新submodule时候会非常频繁的拉不下来，考虑到GAMES104学生新人比较多，所以就移除了submodule，采用了内置package的方式。
